profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/binder/BaseProfitBricksRequestBinder.java

```diff
@@ -16,20 +16,18 @@
  */
 package org.apache.jclouds.profitbricks.rest.binder;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
 import java.util.Map;
-
 import org.jclouds.http.HttpRequest;
 import org.jclouds.rest.MapBinder;
-
 import com.google.common.base.Strings;
-
 import org.jclouds.io.MutableContentMetadata;
 import org.jclouds.io.payloads.BaseMutableContentMetadata;
+import static com.google.common.base.Preconditions.checkNotNull;
 
 public abstract class BaseProfitBricksRequestBinder<T> implements MapBinder {
 
+   protected final String endpointPath = "/rest/v2";
```

Comment:
Don't hardcode this. Extract it from the provider metadata. To do so:

* Add a parameter to the constructor: `Supplier<URI> endpointSupplier`. This will provide the endpoint configured in the metadata.
* Modify subclasses to inject the endpoint supplier, by adding the supplier parameter as: `@Provider Supplier<URI> endpointSupplier` (and making sure the constructor is annotated with `@Inject`). Note that subclasses add the qualifier annotation to get the right supplier injected. It would be redundant in the parent class as it is abstract and shouldn't care about implementations using injection or not.